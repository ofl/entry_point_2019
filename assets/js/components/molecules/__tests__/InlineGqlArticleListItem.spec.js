import { shallowMount, RouterLinkStub } from "@vue/test-utils";
import InlineGqlArticleListItem from "../InlineGqlArticleListItem.vue";
import { fakeArticle } from "../../../test/fake-data";

describe("InlineGqlArticleListItem", () => {
  test("is a Vue instance", () => {
    const wrapper = shallowMount(InlineGqlArticleListItem, {
      propsData: { article: fakeArticle },
      stubs: { RouterLink: RouterLinkStub }
    });

    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders an article", () => {
    const wrapper = shallowMount(InlineGqlArticleListItem, {
      propsData: { article: fakeArticle },
      stubs: { RouterLink: RouterLinkStub }
    });

    expect(wrapper.find("h4").text()).toBe(fakeArticle.title);
  });

  test("renders an article", () => {
    const wrapper = shallowMount(InlineGqlArticleListItem, {
      propsData: { article: fakeArticle },
      stubs: { RouterLink: RouterLinkStub }
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  test("avatarUrl is article author avatar url", () => {
    const wrapper = shallowMount(InlineGqlArticleListItem, {
      propsData: { article: fakeArticle },
      stubs: { RouterLink: RouterLinkStub }
    });

    expect(wrapper.vm.avatarUrl).toBe(fakeArticle.user.avatarUrl);
  });
});
