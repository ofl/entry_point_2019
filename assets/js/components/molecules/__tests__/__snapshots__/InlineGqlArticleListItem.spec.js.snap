// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineGqlArticleListItem renders an article 1`] = `
<article
  class="post"
>
  <a>
    <h4>
      Hello
    </h4>
  </a>
   
  <div
    class="media"
  >
    <div
      class="media-left"
    >
      <p
        class="image is-32x32"
      >
        <img
          src="https://foo.com/br.png"
        />
      </p>
    </div>
     
    <div
      class="media-content"
    >
      <div
        class="content"
      >
        <p>
          <a
            href="#"
          >
             Jhon 
          </a>
          2019-01-01 Â  
          <span
            class="tag"
          >
             Question 
          </span>
        </p>
      </div>
    </div>
     
    <div
      class="media-right"
    >
      <span
        class="has-text-grey-light"
      >
        <i
          class="fa fa-comments"
        />
         1
        
      </span>
       
      <span
        class="has-text-grey-light"
      >
        <i
          class="fa fa-heart"
        />
         1
        
      </span>
    </div>
  </div>
</article>
`;
